clarity/src/vm/analysis/types.rs

```diff
@@ -187,6 +187,21 @@ impl ContractAnalysis {
         self.defined_traits.get(name)
     }
 
+    /// Canonicalize all outward-facing types in the contract analysis.
+    pub fn canonicalize_types(&mut self, epoch: &StacksEpochId) {
+        for (_, function_type) in self.public_function_types.iter_mut() {
+            *function_type = function_type.canonicalize(epoch);
+        }
+        for (_, function_type) in self.read_only_function_types.iter_mut() {
+            *function_type = function_type.canonicalize(epoch);
+        }
+        for (_, trait_definition) in self.defined_traits.iter_mut() {
+            for (_, function_signature) in trait_definition.iter_mut() {
+                *function_signature = function_signature.canonicalize(epoch);
+            }
+        }
```

Comment:
How defensive do we want to be here?  Should we canonicalize _everything_, even if it's not outward facing?  Asking because in pox-2, for example, there's a private function that the node will call directly.

Reply:
I'm not strongly opposed to canonicalizing everything to be safe. I can make that change today.

Reply:
Added in 70d72a992.